Date: Jan 20, 2016 7:16 AM

From: 23d20db6af215695fb6fdaf7098b6aba8c01eeb2 Jan 21, 2016 2:46 AM

Subject: [PATCH] ARRISXB3-4105: Bringing libupnp on Yocto to 1.6.18 on par with Legacy.

     WECB is not functional on Yocto now. In order to isolate
     the issue in Arris, this step is intended to reduce the
     number of unknowns.

Source: COMCAST


Upstream-Status: Pending

Signed-off-by: Ramki Ananthakrishnan <ramki_ananthakrishnan@cable.comcast.com>
---

--- a/build-aux/ltmain.sh	2013-02-19 17:38:50.831930000 +0800
+++ b/build-aux/ltmain.sh	2013-02-19 17:58:15.643132000 +0800
@@ -6504,8 +6504,14 @@
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
 	  else
-	    dir="$lt_sysroot$libdir"
-	    absdir="$lt_sysroot$libdir"
+	    # Adding 'libdir' from the .la file to our library search paths
+	    # breaks crosscompilation horribly.  We cheat here and don't add
+	    # it, instead adding the path where we found the .la.  -CL
+	    dir="$abs_ladir"
+	    absdir="$abs_ladir"
+	    libdir="$abs_ladir"
+	    #dir="$lt_sysroot$libdir"
+	    #absdir="$lt_sysroot$libdir"
 	  fi
 	  test "X$hardcode_automatic" = Xyes && avoidtemprpath=yes
 	else
@@ -6904,7 +6910,7 @@
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      #add_dir="-L$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
@@ -7059,7 +7065,18 @@
 		  fi
 		  ;;
 		*)
-		  path="-L$absdir/$objdir"
+		  #path="-L$absdir/$objdir"
+                    # OE sets installed=no in staging. We need to look in $objdir and $absdir,
+                   # preferring $objdir. RP 31/04/2008
+                  if test -f "$absdir/$objdir/$depdepl" ; then
+		    depdepl="$absdir/$objdir/$depdepl"
+		    path="-L$absdir/$objdir"
+                  elif test -f "$absdir/$depdepl" ; then
+		    depdepl="$absdir/$depdepl"
+		    path="-L$absdir"
+                  else
+		    path="-L$absdir/$objdir"
+                  fi
 		  ;;
 		esac
 		else
